src/rules/tictactoepoker/tictactoepoker.ts

```diff
@@ -0,0 +1,752 @@
+    Flush = 4,
+    StraightFlush = 5,
+}
+
+// Score type.
+interface ScoreType {
+    combination_type: CombinationType,
+    rank_card?: Card,
+}
+
+// Returns the score value for the given score type.
+function compute_score_value(score_type: ScoreType): number {
+    let rank_value = 0;
+    if (score_type.rank_card) {
+        rank_value = score_type.rank_card.get_rank_value();
+    }
+
+    switch(score_type.combination_type) {
+        case CombinationType.StraightFlush:
+            return 1000 + 30 * rank_value;
+        case CombinationType.Flush:
+            return 750 + 15 * rank_value;
+        case CombinationType.Straight:
+            return 500 + 10 * rank_value;
+        case CombinationType.ThreeOfAKind:
+            return 250 + 5 * rank_value;
+        case CombinationType.Pair:
+            return 100 + rank_value;
+        case CombinationType.none:
+            return 0;
+    }
+
+    throw("should not get here");
+}
+
+// Represents an explanation for how a single scoring line (row, column, or diagonal)
+export class ScoreExplanation {
+    private readonly score_type: ScoreType
+    private readonly where: string
+    private readonly score_value: number;
+
+    constructor(score_type: ScoreType, where: string) {
+        this.score_type = score_type;
+        this.where = where;
+        this.score_value = compute_score_value(score_type);
+    }
+
+    public get_score_value(): number {
+        return this.score_value;
+    }
+
+    public toString(): string {
+        return CombinationType[this.score_type.combination_type] + " in " + this.where + " " +
+            "with ranking card " + this.score_type.rank_card.toString() + " worth " +
+            this.score_value + " points";
+    }
+}
+
+export interface ScoreWithExplanations {
+    score: number,
+    explanations: ScoreExplanation[]
+}
+
+// Return score type for the given three cards (some of all of which may be empty cards)
+function compute_score(a: Card, b: Card, c: Card): ScoreType {
+    // Get all normal cards
+    let cards = [];
+    if (a.is_normal()) { cards.push(a); }
+    if (b.is_normal()) { cards.push(b); }
+    if (c.is_normal()) { cards.push(c); }
+
+    // Compute score for an array of cards, where the Aces have been resolved to
+    // either 1 or 14
+    const compute_no_ace = ((cards) => {
+
+        if (cards.length == 3) {
+
+            // Flush (all same suit)
+            const is_flush: boolean = (() => {
+                // All the same suit.
+                let suits = {};
+                for (let card of cards) {
+                    suits[card.get_suit()] = true;
+                }
+                if (Object.keys(suits).length != 1) {
+                    return false;
+                }
+
+                return true;
+            })();
+
+
+            // Straight (just three consecutive cards)
+            const is_straight: boolean = (() => {
+                if (cards[0].get_value() + 1 == cards[1].get_value() &&
```

Comment:
i'm surprise eslint didn't complain, but use === for strict equality (no implicit cast)

Reply:
ESLint doesn't complain. What is being implicitly casted?

Reply:
nothing here, but it's just good practice. == will allow for implicit casting if types of operands mismatch, e.g the predicate (1 == true) evaluates to true.